LSPS2/README.md

```diff
@@ -0,0 +1,955 @@
+> published with `announce_channel = true`.
+
+Clients MAY reject the channel open attempt if any other
+parameters or options are not desired by the client (for example,
+it MAY reject non-zero channel reserve requirements imposed on
+the client by the LSP).
+Rejection is via the `error` Lightning BOLT message.
+
+LSPs MUST fail the payment with `unknown_next_peer` in case the
+client rejects the channel open via an `error`.
+
+If the client disconnects before the LSP can receive `funding_signed`
+from the client, the LSP MUST fail the incoming payment parts with
+`temporary_channel_failure`, then return to the previous step (waiting
+for payment).
+
+> **Rationale** The client might have crashed at this point, and
+> recovery might take an inordinate amount of time.
+> However, use of the `temporary_channel_failure` error informs the
+> payer that the error is temporary and it can retry later, if the
+> client is online at that point.
+>
+> Client disconnection before the LSP can receive `funding_signed`
+> will prevent the LSP from broadcasting the funding transaction
+> and the LSP may re-allocate the UTXOs spent in the funding
+> transaction for a different payment.
+> Thus, the correct behavior is to abort the payment, so that the
+> LSP funds may be re-used in a new channel to the same client if
+> the payer retries, or so that the LSP funds may be re-used for a
+> different client if the payer gives up on paying the client.
+
+Clients MUST NOT expect that `jit_channel_scid` is the same as
+the `alias` SCID.
+Clients MUST NOT expect that `jit_channel_scid` is different from
+the `alias` SCID.
+
+Depending on the `client_trusts_lsp` from the `lsps2.buy` result,
+after the client has sent `funding_signed`:
+
+* If `client_trusts_lsp` is specified and `true`:
+  * The LSP MAY wait for the client to send the preimage to the
+    incoming payment via a `update_fulfill_htlc` before broadcasting
+    the funding transaction.
+  * The client MUST NOT wait for the LSP to broadcast the funding
+    transaction before sending the preimage via a
+    `update_fulfill_htlc`.
+  * The client and LSP MUST immediately send `channel_ready`.
+* If `client_trusts_lsp` is unspecified or `false`:
+  * The client MAY wait for the funding transaction to appear in
+    its mempool or the mempool of a trusted node, or confirmed in a
+    block with any depth, AND confirm that the funding transaction
+    output is correct, before sending `channel_ready`.
+  * The LSP MUST immediately broadcast the channel funding
+    transaction and send `channel_ready`.
+  * The client MAY immediately send `channel_ready` (i.e. the
+    client trusts the LSP, even though the LSP does not require
+    that trust).
+
+> **Rationale** As mentioned in the section "Trust Models" above,
+> in LSP trusts client model, the client is allowed to choose its
+> risk appetite; it may accept 0 confirmations, 1, 3, 6, 100, or
+> however many it believes would be safe with the particular LSP.
```

Comment:
I think there's a difference between 'allowed by this protocol', and 'possible to do so'. The client should not wait for confirmations. It should buy an inbound channel through LSPS1 or any other way if it needs confirmations. It should not allow locking up the liquidity in the network for a long amount of time.

Reply:
Without confirmations and without a punishable LSPS3, the client has to trust the LSP.  The mere fact that the client is allowed to wait for confirmations, without any spec verbiage disallowing (no "client SHOULD NOT wait for confirmations" unless `client_trusts_lsps` exists and is `true`) is important here outside of technical considerations.  the client is explicitly allowed to do this (i.e. harm the network for its own benefit) as that prevents the LSP from being able to cheat arbitrary clients.